src/app/tabs/camera.cpp

```diff
@@ -1,135 +1,361 @@
+    QCheckBox *network_toggle = new QCheckBox("Network", this);
+    network_toggle->setFont(Theme::font_14);
+    connect(network_toggle, &QCheckBox::toggled, [this, network_cam, local_cams](bool checked) {
+        network_cam->setVisible(checked);
+        local_cams->setVisible(!checked);
+        this->status->setText("");
+    this->config->set_cam_is_network(checked);
+    });
+    network_toggle->setChecked(this->config->get_cam_is_network());
+
+    layout->addStretch();
+    layout->addWidget(label, 0, Qt::AlignCenter);
+    layout->addStretch();
+    layout->addWidget(local_cams);
+    layout->addWidget(network_cam);
+    layout->addWidget(this->status, 0, Qt::AlignCenter);
+    layout->addWidget(this->connect_button(), 0, Qt::AlignCenter);
+    layout->addStretch();
+    layout->addWidget(network_toggle, 1, Qt::AlignLeft);
+
+    return widget;
+}
+
+QWidget *CameraTab::local_camera_widget()
+{
+    QWidget *widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->setSpacing(0);
+
+    QPushButton *disconnect = new QPushButton(widget);
+    disconnect->setFlat(true);
+    disconnect->setIconSize(Theme::icon_16);
+    connect(disconnect, &QPushButton::clicked, [this]() {
+        this->status->setText(QString());
+        this->local_cam->stop();
+        this->local_cam->unload();
+        this->local_cam->deleteLater();
+        this->local_cam = nullptr;
+    });
+    this->theme->add_button_icon("close", disconnect);
+    layout->addWidget(disconnect, 0, Qt::AlignRight);
+
+    this->local_video_widget = new QCameraViewfinder(widget);
+    layout->addWidget(this->local_video_widget);
+
+    return widget;
+}
+
+QWidget *CameraTab::network_camera_widget()
+{
+    QWidget *widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+    layout->setContentsMargins(0, 0, 0, 0);
+    layout->setSpacing(0);
+
+    QPushButton *disconnect = new QPushButton(widget);
+    disconnect->setFlat(true);
+    disconnect->setIconSize(Theme::icon_16);
+    connect(disconnect, &QPushButton::clicked, [this]() {
+        this->player->setMedia(QUrl());
+        this->status->setText(QString());
+        this->player->stop();
+    });
+    this->theme->add_button_icon("close", disconnect);
+    layout->addWidget(disconnect, 0, Qt::AlignRight);
+
+    QVideoWidget *video = new QVideoWidget(widget);
+    this->player->setVideoOutput(video);
+    layout->addWidget(video);
+
+    return widget;
+}
+
+
+QPushButton *CameraTab::connect_button()
+{
+    QPushButton *connect_button = new QPushButton("connect", this);
+    connect_button->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+    connect_button->setFont(Theme::font_14);
+    connect_button->setFlat(true);
+    connect_button->setIconSize(Theme::icon_36);
+    connect(connect_button, &QPushButton::clicked, [this]() {
+        this->status->setText("");
```

Comment:
set to `QString()` here

Reply:
That's identical but harder to read. Even the examples in the Qt doc use plain strings.

Reply:
not the same, `QString()` is a null string while `""` is an empty string, and maybe we want to be able to differentiate between the two down the road?

The reason i pointed it out is because the network cam does `QString()`

looking at QLabel tho it look like it has a `clear` function, which imo would be the most straightforward

Reply:
Ok, done.